mlflow/data.py

```diff
@@ -27,33 +32,28 @@ def _fetch_dbfs(uri, local_path):
 def _fetch_s3(uri, local_path):
     import boto3
     print("=== Downloading S3 object %s to local path %s ===" % (uri, os.path.abspath(local_path)))
-    (bucket, s3_path) = parse_s3_uri(uri)
+    (bucket, s3_path) = parse_simple_uri(uri, ["s3"])
     boto3.client('s3').download_file(bucket, s3_path, local_path)
 
 
 def _fetch_gs(uri, local_path):
     from google.cloud import storage
     print("=== Downloading GCS file %s to local path %s ===" % (uri, os.path.abspath(local_path)))
-    (bucket, gs_path) = parse_gs_uri(uri)
+    (bucket, gs_path) = parse_simple_uri(uri, ["gs"])
     storage.Client().bucket(bucket).blob(gs_path).download_to_filename(local_path)
 
 
-def parse_s3_uri(uri):
-    """Parse an S3 URI, returning (bucket, path)"""
-    parsed = urllib.parse.urlparse(uri)
-    if parsed.scheme != "s3":
-        raise Exception("Not an S3 URI: %s" % uri)
-    path = parsed.path
-    if path.startswith('/'):
-        path = path[1:]
-    return parsed.netloc, path
+def _fetch_hdfs(uri, local_path):
+    print("=== Downloading HDFS file %s to local path %s ===" % (uri, os.path.abspath(local_path)))
+    parse_simple_uri(uri, ["hdfs", "viewfs"])
+    hdfs_repo = HdfsArtifactRepository(uri)
+    hdfs_repo._download_file(uri, local_path)
```

Comment:
Can you use

```
  hdfs_repo.download_artifacts(uri, local_path)
```

If you need to special handling for the trailing `/` can you do it here in the before calling `download_artifacts`?